Fix shellcheck errors in scripts (part 3)

MuteSpirit opened the pull request:
Fixing shellcheck issues in 3rd part of scripts mentioned in task #915

Description of fixed shellcheck issues you may found in commit comments

rmelotte commented on framework/platform/bpl/uci/scripts/platform_daemon.sh:
```diff
@@ -23,7 +23,7 @@ repeater_to_ap()
 
     # Kill DHCP client
-    ps -A | grep [d]hcp | awk '{print $1;}' | xargs kill
+    pgrep [d]hcp | xargs kill
```
This change is only possible if pgrep is available on the target platform.
Please check whether it's the case or not (@kantera800 ? @tomereli ?)

MuteSpirit replied:
I've checked that `pgrep` is present on RAX40 board.
     And absent `pkill` because otherwise i'd use it instead of.
I can check `pgrep` presence in built images for another platforms.

adam1985d replied:
I've verified existence of `pgrep` on the UGW and RDKB platforms
`pkill` on the other hand exists in RDKB but not in UGW
@tomereli @kantera800

tomereli replied:
In this case (UGW is much harder to change since it is a real product) - lets not use `pkill`.

MuteSpirit replied:
I've checked that `pgrep` is present in nightly build images for `turris-omnia` and `netgear-rax40`.
Also checked that config option `CONFIG_BUSYBOX_DEFAULT_PGREP=y` is present in `.config` OpenWrt file for device `glinet-b1300`.

So using `pgrep` looks reasonable

rmelotte commented on framework/platform/bpl/uci/scripts/platform_daemon.sh:
```diff
@@ -54,7 +54,7 @@ static_init()
     # Kill DHCP server
     echo "Kill DHCP server"
-    ps -A | grep [d]ns | awk '{print $1;}' | xargs kill
+    pgrep [d]ns | xargs kill
```
Same comment, please make sure pgrep is already available on the target platform.

rmelotte replied:
> In fact, if pgrep is available, so will be pkill since they're both part of procps.

Not necessarily. At the moment on the rax40 for example, Busybox has a variant of pgrep, but no pkill at all.

tomereli replied:
This we can change - we decide what to have in the RAX40 busybox configuration, I added history for example. John has no problem that we add more capabilities to prplWrt.

adam1985d replied:
see my reply at:
https://github.com/prplfoundation/prplMesh/pull/1133#discussion_r409283040

rmelotte commented on tools/docker/static-analysis/cppcheck.sh:
```diff
@@ -1,22 +1,14 @@
-    -I$rootdir/common/beerocks/bcl/include \
-    -I$rootdir/common/beerocks/bwl/include"
+INCLUDES=()
+while IFS='' read -r line; do INCLUDES+=("-I$line"); done < <(find -L "$rootdir/common" "$rootdir/framework" -type d -name include)
```
Nice! We're less likely to forget to add new includes over time (or to remove old ones)...

rmelotte reviewed (approved):
Please double-check `pgrep` is available on every platform that use `platform_daemon.sh`.
Other than that only nitpicks :+1: .

Commit:
scripts: cppcheck.sh: fix shellcheck issues

Fix next shellcheck issues in cppcheck.sh:
  https://www.shellcheck.net/wiki/SC1090 -- Can't follow non-constant source. Use a directive to specify location.
  https://www.shellcheck.net/wiki/SC2164 -- Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing and word splitting.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: cppcheck.sh: find include folders in runtime instead of hard-code

"cppcheck.sh" contains list of hard-coded paths to folders with header
files. That list should be updated each time of add new "include"
folder.

Replace list of folders with "find" command seaching directories with
name "include" inside "${rootdir}/framework" and "${rootdir}/common"

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: stop.sh: fix shellcheck issues SC2039, SC2164

Add handle error during parsing cmdline arguments

Fix shellcheck issues in tools/docker/stop.sh:
  https://www.shellcheck.net/wiki/SC2039 -- In POSIX sh, 'local' is undefined.
  https://www.shellcheck.net/wiki/SC2164 -- Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: tools/docker/test.sh: fix shellcheck issues SC2164, SC2068, SC1090, SC2034

Fix shellcheck issues in tools/docker/test.sh:
  https://www.shellcheck.net/wiki/SC2164 -- Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
  https://www.shellcheck.net/wiki/SC2068 -- Double quote to prevent globbing and word splitting.
  https://www.shellcheck.net/wiki/SC1090 -- Can't follow non-constant source. Use a directive to specify location.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: run.sh: fix shellcheck issue SC2086

Fix shellcheck issue in tools/docker/run.sh:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing and word splitting.

Change '/bin/sh' -> '/bin/bash' in shebang

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: bitbake.sh: fix shellcheck issue SC2086

Fix shellcheck issue in bitbake.sh:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing and word splitting.

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
scripts: platform_daemon.sh: fix shellcheck issues SC2046, SC2009

Fix shellcheck issues "framework/platform/bpl/uci/scripts/platform_daemon.sh":
  https://www.shellcheck.net/wiki/SC2046 -- Quote this to prevent word splitting.
  https://www.shellcheck.net/wiki/SC2009 -- Consider using pgrep instead of grepping ps output.

Use "pgrep" instead of "ps -A | grep"

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>
